neo/rawio/plexon2rawio/plexon2rawio.py

```diff
@@ -128,7 +132,6 @@ def __init__(self, filename, pl2_dll_file_path=None):
 
         self.pl2reader = PyPL2FileReader(pl2_dll_file_path=pl2_dll_file_path)
 
-        reading_attempts = 10
```

Comment:
Is it possible that you have a good attempt in the middle and bad attempt at the end of the range that then messes up the attempts such that you just have to guess when to end on a "good attempt". Ie is there a smarter way for us to check like

for i in range(num_attempts):
read
if read is good
break

Reply:
That's how it is right now. If the handle is good you exist the loop:

https://github.com/NeuralEnsemble/python-neo/blob/221befab6c1d90567107d84bc58ad233374614b8/neo/rawio/plexon2rawio/plexon2rawio.py#L139-L141

In other words, it is first success out.

Reply:
sorry I missed that line. Sorry didn't expand the diff.

Reply:
It is a good suggestion though!

Reply:
And so it silently fails? We have yet to see any stochastic failures of our test files so far. It's just so weird for it to fail. Has plexon commented on this?

Reply:
You did at the beginning before we made this patch:

https://github.com/NeuralEnsemble/python-neo/pull/1495#issuecomment-2184193508

The re-attempt mechanism solves most of the case almost all of the time except when it does not. For some reason it has been failing for neuroconv's docstring tests.

Reply:
I meant test failures since your PR. It seems like the 10 tries has always worked for us thus far. It seems weird that the flakiness is so flaky, but yeah based on the discussion with plexon there they also seem to be a bit confused by this issue. So I'm fine with just letting the person try as many times as they want.

Reply:
Yeah, it is kind of fishy that it always fails in our doctest. But if I discover that that indeed is the case that more attempst solve the problem and that different envs require different attemps I don't want to have to create a new PR every time so better exposed it.

It is unlikely that we find a root case solution if the plexon people also are suprised about it because it is probably a wine dllc related problem.